### PR TITLE
br_if-to-table

### PR DESCRIPTION
**Pull request by kripken**
Note: unrelated to [farm-to-table](https://en.wikipedia.org/wiki/Farm-to-table).

Implements #1309: subsequent `br_if`s that compare the same value to various constants are converted into a `br_table` in a block,
````
(br_if $x (i32.eq (get_local $a) (i32.const 0)))
(br_if $y (i32.eq (get_local $a) (i32.const 1)))
(br_if $z (i32.eq (get_local $a) (i32.const 2)))
  ==>
(block $tablify
 (br_table $x $y $z $tablify
  (get_local $a)
 )
)
````

The constants for when to apply this (e.g., not if the range of values would make a huge jump table) are fairly conservative, I think, but hard to tell. Probably should be tweaked based on our experience with the pass in practice later on.

cc @dcodeIO

### CODE REVIEW

**Comment by kripken** [comment]
This PR also includes a fix for a missing `finalize` on a `makeBlock` variant, that this happened to uncover.

File: src/passes/RemoveUnusedBrs.cpp [hunk 1]
```diff
@@ -556,11 +562,169 @@ struct RemoveUnusedBrs : public WalkerPass<PostWalker<RemoveUnusedBrs>> {
...
+        // returns the br_if if so, or nullptr otherwise
+        auto getProperBrIf = [](Expression* curr) -> Break*{
+          auto* br = curr->dynCast<Break>();
+          if (!br) return nullptr;
+          if (!br->condition || br->value) return nullptr;
+          if (br->type != none) return nullptr; // no value, so can be unreachable or none. ignore unreachable ones, dce will clean it up
+          auto* binary = br->condition->dynCast<Binary>();
+          if (!binary) return nullptr;
+          if (binary->op != EqInt32) return nullptr;
+          auto* c = binary->right->dynCast<Const>();
```
Comment (dcodeIO) [inline src/passes/RemoveUnusedBrs.cpp]: Might it make sense to test for the inverse as well, i.e., the constant in `left` and the condition in `right`?

Comment (kripken) [reply src/passes/RemoveUnusedBrs.cpp]: The optimize-instructions pass will canonicalize things so a comparison of a const to anything else leaves the const on the right. And it makes sense to run that pass before this one anyhow, I think?

Comment (dcodeIO) [reply src/passes/RemoveUnusedBrs.cpp]: I see, didn't know that. How little I know after all :)

Comment (kripken) [reply src/passes/RemoveUnusedBrs.cpp]: Heh, well, it's an internal optimizer detail. Should probably be documented somehow.

**Comment by kripken** [comment]
Looks like no concerns with this new optimization, landing.

**Pull request merged**